### DISCUSSION TITLE
Zabezpieczanie databricks - szczególnie reguły allow in/out dla VirtualNetwork

### DISCUSSION THREAD

**Discussion by maciej-skorupka** [AzureCommunityPL/group#432] [Pomoc] [answered]
Hej,

Kombinuję z zabezpieczaniem databricksa siedzącego w moim VNET. Jak się okazuje databricks wsadzony do mojego VNETu potrzebuje różnych reguł (https://docs.microsoft.com/pl-pl/azure/databricks/administration-guide/cloud-configurations/azure/vnet-inject#network-security-group-rules-for-workspaces-created-after-january-13-2020), które zgodnie z moim rozumieniem pozwalają na dowolne wyjście do innych moich VNETów i wejście z dowolnego z nich. 

Dodam, że nie wpisanie tych reguł do NSG powoduje, że databricks sam je dodaje. Nawet jak dodam je z niskim priorytetem, to databricks je doda z wyższym. Jak dodam blokowanie tego z priorytetem 100, to databricks przesunie moją regułę na wyższy numerek i doda swoje na 100. Innymi słowy brzmi jak bym nie był w stanie tego przyciąć inaczej niż stawiając gdzieś dalej firewalla i routując przez niego wszystko. 

Znacie inne sposoby? Może źle rozumiem allow in/out na VirtualNetwork i tak naprawdę nie oznacza to otwarcia ruchu do dowolnego routowalnego VNETu.

**Comment by piotrgwiazda** [comment, marked as answer]
Sereis tag VirtualNetwork oznacza bieżący vNet. Sprawa zaczyna się komplikować jak dołączasz inną sieć przez peering. Wtedy to zależy od tego co wybierzesz w opcjach allow traffic... [Ta strona dokumentacji](https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-manage-peering#view-or-change-peering-settings) dość dokładnie opisuje co się dzieje z tym tagiem w zależności od opcji peeringu.

**Reply by maciej-skorupka** [reply]
Czyli w zasadzie jeśli databricks wstawia taką regułę i mam peeringi, to muszę postawić firewall po drodze, jeśli chcę tym jakkolwiek zarządzić?

**Reply by piotrgwiazda** [reply]
A jak masz ustawione peeringi? Jeśli wyłączysz "Allow all traffic" to wtedy regukujesz ruch za pomocą NSG, bo reguła VirtualNetwork nie będzie "łapać" sieci speerowanych.

**Reply by maciej-skorupka** [reply]
Dzięki, super sugestie.
